platform/engineering/design-docs/flag-for-manual-review.md

```diff
@@ -0,0 +1,112 @@
+# Flag for manual review process
+
+**Author(s):** Brooks Johnson (brooks@adhoc.team)
+**Last Updated:** February 20, 2020  
+**Status:** **Draft** | In Review | Approved  
+**Approvers:** _Rian Fowler_ \[ \], _Andrew Gunsch_ \[ \], ...  
+
+
+
+## Overview
+
+### Objective
+
+The goal is to improve our review process which by adding automated processes which will alert a review team when a manual review is required.
+These changes are meant to work alongside the Codeowners changes which will give each app team primary responsibility for reviewing their code.
+
+
+Intended audience: VSP team
+
+### Background
+
+From the perspective of a VFS engineer, the current review process involves:
+
+1. Creating a Draft PR with appropriate changes
+1. Getting a peer review
+    - Making any changes until the review is approving
+1. Marking the PR as `Ready for Review`
+1. Getting approval from a member of the VSP *frontend-review-group*
+
+The final approval from the review group is often not helpful, and it slows the process down.
+This frustration often leads to VFS teams ignoring this process and they will directly ask for a VSP review before their PR is ready.
+
+### High Level Design
+
+The plan is to use a CI process to run a script each time a change gets pushed to a PR.
+This script will look for anything that should trigger a manual review by the VSP review group, and alert them if it finds something.
+
+## Specifics
+
+### Detailed Design
+
+We will use a script that can be run from a job in CircleCI.
+The script will:
+
+- Diff the current PR branch against master
+- Make a pass which will mark any additions with the filename and position in the diff
+- Search the processed diff for occurrences of anything that should warrant a manual review
+- Remove any items from the list of offense additions that have already been commented on by the review bot
+- If there are any offenses which haven't been commented on, leave a comment and request a review from the *frontend-review-group*
+
+This will rely on the Github API as well as some environment variables provided by Circle CI.
+
+### Code Location
+
+- `script/pr-check.js`
+- `.circleci/config.yml`
+
+### Testing Plan
+I will run the script locally in addition to having Circle run it.
+
+### Logging
+The script will log to the console:
+- Whether or not the branch it is running on has an associated PR
+- If there are no offenses that would warrant a comment
+- Any previous comments made that do not need to be made again
+- Any new comments that will be made
+
+### Debugging
+
+If a user sets up the proper environment variables locally, the script can be run in a local environment and the logging output can be viewed.  Otherwise we will rely on viewing CircleCI logs.
```

Comment:
I put the environment variables in the same doc with the list of triggers and linked to it.  I'm not convinced that that is the best place,  so that may change as I clean things up.